### PR TITLE
Allow users to set the waiting time when displaying a preview image using an external viewer

### LINKED ISSUE #1061: Fail to open images in external viewer programs if running PyGMT in a Python script
**Description of the problem**

Here is a simple Python script to reproduce the issue. 
```python
import pygmt

fig = pygmt.Figure()
fig.basemap(region=[0, 10, 0, 10], projection="X10c/10c", frame="+tFigure1")
print("Openning Figure1")
fig.show(method="external")
```
When I run the codes in a Python/IPython interpreter, it works well and can open the preview image. However, if I save it in a Python script and run it using `python test.py`, sometimes it can open the image but sometimes it fails and gives an error like (I'm using macOS):

image

The reason is simple, on macOS, we call the `open` program to open the image. Because we can't detect the status of the external viewer, the Python script continues to execute the code and exit. When the script exits, Python will call the destructor of the Figure class (i.e., `Figure.__del__` method), which deletes the preview directory and also the preview image.

So, the preview image will show if the `open` program opens the image before Python deletes it, and will fail to open the image if it's already deleted.

---

The following script is similar to the above one, but instead, it plots and opens two images.
```
import pygmt

fig = pygmt.Figure()
fig.basemap(region=[0, 10, 0, 10], projection="X10c/10c", frame="+tFigure1")
print("Openning Figure1")
fig.show(method="external")

fig = pygmt.Figure()
fig.basemap(region=[0, 10, 0, 10], projection="X10c/10c", frame="+tFigure2")
print("Openning Figure2")
fig.show(method="external")
``` 

Again, the script works well in a Python/IPython interpreter, but usually can open Figure1 and can't open Figure2 when running a script. Figure1 can be successfully opened because plotting the second figure take some time, so the first `Figure.show()` call can open Figure1 before the program exits.

---

The only solution I can figure out is adding a short pause after calling the external viewer, i.e., 
adding `time.sleep(0.5)` at the end of the `launch_external_viewer(fname)` function.

https://github.com/GenericMappingTools/pygmt/blob/1fb9e2e4e523baaf3aaedb90d71872ede6d03bb7/pygmt/helpers/utils.py#L195

It works well for me, and should also work for Linux, but I'm not sure about Windows. The 0.5 second pause time is empirical, and may not work for slow computers.

I opened the PR #1062 with the above solution, so that anyone can test it with/without the pause.

### PR DESCRIPTION
**Pull request by seisman**
**Description of proposed changes**

See https://github.com/GenericMappingTools/pygmt/issues/1061 for the detailed issue report and discussions.

This PR adds the `waiting` parameter to the `launch_external_viewer` function. The default value is 0, so no waiting by default.

As mentioned in https://github.com/GenericMappingTools/pygmt/issues/1061#issuecomment-960374177, the `launch_external_viewer` function is used at two places, one in the `Figure.show()` function, another in the `Figure.savefig()` function. 

- The function call in the `Figure.show()` opens the preview image in the temporary directory, thus we need to set the waiting time to a non-zero value (`waiting=0.5` works for most users) so the external viewer can open the preview image before the PyGMT session exits.
- The function call in the `Figure.savefig()` opens the image that was already saved in the current directory or elsewhere, and won't be deleted when the PyGMT session exits. Thus there is no need to set the waiting time here.

Fixes #1061.

**Reminders**

- [ ] Run `make format` and `make check` to make sure the code follows the style guide.
- [ ] Add tests for new features or tests that would have caught the bug that you're fixing.
- [ ] Add new public functions/methods/classes to `doc/api/index.rst`.
- [ ] Write detailed docstrings for all functions/methods.
- [ ] If adding new functionality, add an example to docstrings or tutorials.

**Slash Commands**

You can write slash commands (`/command`) in the first line of a comment to perform
specific operations. Supported slash commands are:

- `/format`: automatically format and lint the code
- `/test-gmt-dev`: run full tests on the latest GMT development version

### CODE REVIEW

**Comment by willschlitzer** [comment]
I assume this change would slow down running multiple calls of `fig.show()` in a notebook environment? If that is the case, I would think it would slow down processes like running a notebook with multiple figures or when building documentation. Could this be changed to apply the waiting time only when `method="external"`?

File: pygmt/figure.py [hunk 1]
```diff
@@ -284,7 +284,12 @@ def show(self, dpi=300, width=500, method=None):
         This is useful when running unit tests and building the documentation
         in consoles without a Graphical User Interface.
 
-        Note that the external viewer does not block the current process.
+        Note that the external viewer does not block the current process, thus
+        it's necessary to suspend the execution of the current process for a
+        short while after launching the external view, so that the preview
+        image won't be deleted before the external viewer tries to open it. Set
+        the ``waiting`` parameter to a larger number if your computer is old
+        and slow.
```
Comment (willschlitzer) [inline pygmt/figure.py:292]:
```suggestion
        the ``waiting`` parameter to a larger number if your computer is slow.
```
I feel like "slow" is enough to let the user know to change the waiting parameter.

File: pygmt/figure.py [hunk 2]
```diff
@@ -298,6 +302,10 @@ def show(self, dpi=300, width=500, method=None):
             - **external**: PDF preview in an external program [default]
             - **notebook**: PNG preview [default in Jupyter notebooks]
             - **none**: Disable image preview
+        waiting : float
+            Suspend the execution of the current process for a given number of
+            seconds after launching an external viewer.
+            Only works if ``method="external"``.
```
Comment (weiji14) [inline pygmt/figure.py:308]:
```suggestion
            seconds after launching an external viewer. [Default is 0.5].
            Only works if ``method="external"``.
```

Comment (seisman) [reply pygmt/figure.py:308]: It's clear that `waiting` has a default value of 0.5 from the function parameter `waiting=0.5`, thus no need to mention it in the docstrings.

**Review by weiji14 — APPROVED** [review]

File: pygmt/figure.py [hunk 3]
```diff
@@ -284,7 +284,11 @@ def show(self, dpi=300, width=500, method=None):
         This is useful when running unit tests and building the documentation
         in consoles without a Graphical User Interface.
 
-        Note that the external viewer does not block the current process.
+        Note that the external viewer does not block the current process, thus
+        it's necessary to suspend the execution of the current process for a
+        short while after launching the external view, so that the preview
```
Comment (seisman) [inline pygmt/figure.py:289]:
```suggestion
        short while after launching the external viewer, so that the preview
```

**Review by michaelgrund — APPROVED** [review]

**Review by willschlitzer — APPROVED** [review]

**Pull request merged**